protocol/consistencychecks.go

```diff
@@ -0,0 +1,265 @@
+// Implements all consistency check operations done by a CONIKS client
+// on data received from a CONIKS directory.
+// These include data binding proof verification,
+// and non-equivocation checks.
+
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+const (
+	invalidProof int = iota
+	proofOfAbsence
+	proofOfInclusion
+)
+
+// ConsistencyChecks stores the current consistency check
+// state of a CONIKS client. This includes the current epoch
+// and SignedTreeRoot value, as well as directory's policies
+// (e.g., wherether the TemporaryBinding extension is being used or not)
+// The client should create a new ConsistencyChecks instance only once,
+// when she registers her binding with a ConiksDirectory.
+// This ConsistencyChecks instance then will be used to verify
+// the returned responses from the ConiksDirectory.
+type ConsistencyChecks struct {
+	CurrentEpoch uint64
+	SavedSTR     []byte
+
+	// extensions settings
+	isUseTBs bool
+	TBs      []*m.TemporaryBinding
+
+	// signing key
+	// TODO: should we pin the signing key in the client? see #47
+	signKey sign.PublicKey
+}
+
+// NewCC creates an instance of ConsistencyChecks using
+// the pinning directory's STR at epoch 0.
```

Comment:
Oh no, that's my bad. You're right, the client should initialize the CC using its last saved STR (which is STR_0 if this is the first time it initializes the CC).